### PR TITLE
Github connector now returns a full group list when no org is specified

### LINKED ISSUE #1102: Github: Groups not getting loaded when no org is specified in config
Dex is not fetching the groups when no org is specified in the config.
The groups scope is specified `&scope=openid%20email%20profile%20groups`

According to the Github Oauth Application in my private profile dex has the permission to read them.
Am i missing something here?

Version: 2.7.1

### PR DESCRIPTION
**Pull request by pivotal-jwinters**
Fixes #1102

If you asked for the `groups` scope but didn't specify any orgs, the github connector would return an empty set of groups. This should now be fixed.

Based on this change we now return a list of groups that looks like the following:

```json
[
    "org-1:team-1",
    "org-1:team-2",
    "org-1:team-3",
    "org-2:team-4",
    "org-3"
]
```

If teams exist, they are scoped to an org, otherwise the org is listed on its own.

### CODE REVIEW

**Comment by pivotal-jwinters** [comment]
Looks like I'm using features in httptest server that we're available in golang 1.8.

**Comment by joshadambell** [comment]
This looks great! I was just about to implement this.

**Comment by joshadambell** [comment]
@ericchiang Can we get this reviewed?

File: connector/github/github.go [hunk 1]
```diff
@@ -596,3 +600,76 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
 
 	return groups, nil
 }
+
```
Comment (joshadambell) [inline connector/github/github.go]: Function comment needed.

File: connector/github/github.go [hunk 2]
```diff
@@ -596,3 +600,76 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
...
+			for _, team := range teams {
+				groups = append(groups, org+":"+team)
+			}
+		}
+	}
+
+	return groups, err
+}
+
+func (c *githubConnector) userOrgs(ctx context.Context, client *http.Client) ([]string, error) {
```
Comment (joshadambell) [inline connector/github/github.go]: Same here.

File: connector/github/github.go [hunk 3]
```diff
@@ -596,3 +600,76 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
...
+		}
+
+		if apiURL == "" {
+			break
+		}
+	}
+
+	return groups, nil
+}
+
```
Comment (joshadambell) [inline connector/github/github.go]: Same here.

File: connector/github/github.go [hunk 4]
```diff
@@ -596,3 +600,76 @@ func (c *githubConnector) teamsForOrg(ctx context.Context, client *http.Client,
...
+	if err != nil {
+		return groups, err
+	}
+
+	for _, org := range orgs {
+		if teams, ok := orgTeams[org]; !ok {
+			groups = append(groups, org)
+		} else {
+			for _, team := range teams {
+				groups = append(groups, org+":"+team)
```
Comment (joshadambell) [inline connector/github/github.go]: Do you want to put this and line 357 as a function so formatting is spread out?

**Comment by pivotal-jwinters** [comment]
@joshadambell updated as per your suggestions. thanks.

**Comment by mrIncompetent** [comment]
@ericchiang Anything missing on this PR?

**Comment by vito** [comment]
I ended up deleting the branch that this PR points to.

We've ended up with a bunch of work on our fork of Dex so I spent a lot of time cleaning things up so that we can at least more easily keep up to date with upstream and keep tabs on any PR-able work on separate branches instead. More info here: https://github.com/concourse/dex/issues/1

We'll probably re-submit this PR (along with our other `pr/*` branches) in the future, against https://github.com/concourse/dex/tree/pr/github-teams-no-org-and-slugs

**Pull request closed without merging**